src/file.cpp

```diff
@@ -15,7 +15,7 @@ template <typename T = Core>
 static T* readRuntimeObject(BinaryReader& reader, const RuntimeHeader& header)
 {
 	auto coreObjectKey = reader.readVarUint();
-	auto object = CoreRegistry::makeCoreInstance(coreObjectKey);
+	auto object = CoreRegistry::makeCoreInstance((int) coreObjectKey);
```

Comment:
Totally fine for now but I wonder if in the future we should change makeCoreInstance to accept and unsigned int. Might impact the generator, so let's not worry about it for now.